app/vectorstore/opensearch_hybrid.py

```diff
@@ -0,0 +1,215 @@
+import os
+import sys
+from typing import List
+from langchain.docstore.document import Document
+
+from dotenv import load_dotenv, find_dotenv
+
+from langchain_community.vectorstores import OpenSearchVectorSearch
+from langchain_openai import OpenAIEmbeddings
+from langchain_huggingface import HuggingFaceEmbeddings
+
+from opensearchpy import OpenSearch, RequestsHttpConnection
+
+import pandas as pd
+
+load_dotenv(override=True)
+
+OPENSEARCH_USER = os.getenv("OPENSEARCH_USER")
+OPENSEARCH_HOST = os.getenv("OPENSEARCH_HOST")
+OPENSEARCH_PORT = os.getenv("OPENSEARCH_PORT")
+OPENSEARCH_PW = os.getenv("OPENSEARCH_INITIAL_ADMIN_PASSWORD")
+OPENSEARCH_BLOG_DATA_INDEX = os.getenv("OPENSEARCH_BLOG_DATA_INDEX")
+HUGGINGFACE_EMBEDDING_MODEL = os.getenv("HUGGINGFACE_EMBEDDING_MODEL")
+
+class OpenSearchHybridSearch:
+    def __init__(
+        self,
+        host: str = OPENSEARCH_HOST,
+        port: str = OPENSEARCH_PORT,
+        user: str = OPENSEARCH_USER,
+        pw: str = OPENSEARCH_PW,
+    ):
+        """초기화 메서드
+        OpenSearchHybridSearch 클래스의 인스턴스를 초기화합니다.
+
+        Args:
+            host (str): OpenSearch 호스트 주소.
+            port (str): OpenSearch 포트 번호.
+            user (str): OpenSearch 사용자 이름.
+            pw (str): OpenSearch 비밀번호.
+        """
+        # HuggingFace를 사용하여 임베딩 모델 설정
+        self.embeddings = HuggingFaceEmbeddings(model_name=HUGGINGFACE_EMBEDDING_MODEL)
+
+        # OpenSearch HTTP 인증 정보 설정
+        self.opensearch_http_auth = (user, pw)
+        # OpenSearch 클라이언트 초기화
+        self.client = OpenSearch(
+            hosts=[{"host": host, "port": port}],
+            http_auth=self.opensearch_http_auth,
+            use_ssl=True,
+            verify_certs=False,
+            connection_class=RequestsHttpConnection,
+        )
+        # OpenSearch URL 구성
+        self.openserach_url = f"https://{host}:{port}"
+        # OpenSearch 벡터 스토어 초기화
+        self.vector_store = OpenSearchVectorSearch(
+            index_name=OPENSEARCH_BLOG_DATA_INDEX,
+            embedding_function=self.embeddings,
+            opensearch_url=self.openserach_url,
+            http_auth=self.opensearch_http_auth,
+            use_ssl=True,
+            verify_certs=False,
+            ssl_show_warn=False,
+        )
```

Comment:
HuggingFaceHub에서 임베딩 모델을 가져오는데 시간이 오래 거릴 수 있습니다!
기본적인 사항들을 체크하고 최대한 마지막에 로드하는 방식이 오류 발생시 확인에 소요되는 시간이 적을 것 같습니다.

```suggestion
        # OpenSearch HTTP 인증 정보 설정
        self.opensearch_http_auth = (user, pw)
        # OpenSearch 클라이언트 초기화
        self.client = OpenSearch(
            hosts=[{"host": host, "port": port}],
            http_auth=self.opensearch_http_auth,
            use_ssl=True,
            verify_certs=False,
            connection_class=RequestsHttpConnection,
        )
        # OpenSearch URL 구성
        self.openserach_url = f"https://{host}:{port}"
        
        
        # HuggingFace를 사용하여 임베딩 모델 설정
        self.embeddings = HuggingFaceEmbeddings(model_name=HUGGINGFACE_EMBEDDING_MODEL)

        # OpenSearch 벡터 스토어 초기화
        self.vector_store = OpenSearchVectorSearch(
            index_name=OPENSEARCH_BLOG_DATA_INDEX,
            embedding_function=self.embeddings,
            opensearch_url=self.openserach_url,
            http_auth=self.opensearch_http_auth,
            use_ssl=True,
            verify_certs=False,
            ssl_show_warn=False,
        )
```

Reply:
[9af4741](https://github.com/f-lab-edu/llm-mlops/pull/3/commits/9af474142c99526736b5660477ea2d20b623a279)에 수정했습니다!